src/backend/columnar/columnar_tableam.c

```diff
@@ -647,6 +632,34 @@ columnar_relation_copy_for_cluster(Relation OldHeap, Relation NewHeap,
 }
 
 
+/*
+ * RelationColumnList returns a list of AttrNumber's for the columns that
+ * are not dropped and specified by attr_needed.
+ */
+static List *
+RelationColumnList(TupleDesc tupdesc, Bitmapset *attr_needed)
```

Comment:
Perhaps a better name, like NeededColumnsList or something. "Relation" feels wrong because it doesn't take a relation.